test_communication/test/publisher_py.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import importlib
+import os
+import sys
+import time
+
+# this is needed to allow import of test_communication messages
+sys.path.insert(0, os.getcwd())
+# this is needed to allow rclpy to be imported from the build folder
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(
+    os.getcwd())), 'rclpy'))
+
+
+def fill_msg(msg, message_name, i):
```

Comment:
Having separate functions for each message type would make this code more readable.